memory-viz/src/automate.ts

```diff
@@ -96,7 +96,8 @@ function drawAutomatedStackFrames(
         "right_margin",
     ]) {
         if (!configuration.hasOwnProperty(req_prop)) {
-            configuration[req_prop] = config.obj_x_padding;
+            configuration[req_prop as keyof DisplaySettings] =
```

Comment:
Okay this is better, but I think we can go even further. Pull out this list into a constant (note that it's repeated into the two functions), and declare the type of it to be an array of `keyof DisplaySettings`.